src/01-general/work-orders.apib

```diff
@@ -0,0 +1,199 @@
+
+### workOrders.drafts [GET /workOrders.drafts]
+
+Return all draft work orders.
+
++ Request (application/json)
+
+    + Attributes (object)
+        + filter (object, optional)
+            + customer (object, required)
+                + type (enum[string], required)
+                    + Members
+                        + contact
+                        + company
+                + id: `09e5d75f-f817-4872-9723-57fbb8ff710d` (string, required)
+        + page (Page, optional)
+        + sort (array, optional)
+            + (object)
+                + field (enum[string], required)
+                    + Members
+                        + datetime
+                + order (enum[string], optional)
+                    + Members
+                        + asc
+            + Default
+                + (object)
+                    + field: `datetime`
+                    + order: `asc`
+
++ Response 200 (application/json)
+
+    + Attributes (object)
+        + data (array[WorkOrderInfo])
+
+### workOrders.list [GET /workOrders.list]
+
+Return all finalized work orders.
+
++ Request (application/json)
+
+    + Attributes (object)
+        + filter (object, optional)
+            + customer (object, required)
+                + type (enum[string], required)
+                    + Members
+                        + contact
+                        + company
+                + id: `09e5d75f-f817-4872-9723-57fbb8ff710d` (string, required)
+        + page (Page, optional)
+        + sort (array, optional)
+            + (object)
+                + field (enum[string], required)
+                    + Members
+                        + datetime
+                + order (enum[string], optional)
+                    + Members
+                        + asc
+            + Default
+                + (object)
+                    + field: `datetime`
+                    + order: `asc`
+
++ Response 200 (application/json)
+
+    + Attributes (object)
+        + data (array[WorkOrderInfo])
+
+### workOrders.info [GET /workOrders.info]
+
+Return info for a single (draft or finalized) work order.
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + id: `4afb0a9c-91c6-49ed-a2e5-ce7c1e3a87fb` (string, required)
+
++ Response 200 (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + data (WorkOrderInfoDetail)
+
+### workOrders.attachImage [POST /workOrders.attachImage]
+
+Attach image to a draft work order.
+
++ Request (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + id: `4afb0a9c-91c6-49ed-a2e5-ce7c1e3a87fb` (string, required)
+        + image_data: `data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAIAAACQkWg2AAAAAXNSR0IArs4c6QAAAVlpVFh0WE1MOmNvbS5hZG9iZS54bXAAAAAAADx4OnhtcG1ldGEgeG1sbnM6eD0iYWRvYmU6bnM6bWV0YS8AAAASUVORK5CYII=` (string, required) - PNG image, base64 encoded
+
++ Response 200 (application/json;charset=utf-8)
+
+    + Attributes (object)
+        + data (object)
```

Comment:
Q: Why would we return an id for that image? Is there such a thing as a workorderImage?
Shouldn't a list of image be part of the workorder detail?

Reply:
Agreeing on the latter, but we do need IDs to be able to delete images from draft WorkOrders.